### DISCUSSION TITLE
how to nest "for" segment  in  match segment

### DISCUSSION THREAD

**Discussion by prehonor** [leptos-rs/leptos#2869] [Q&A]
view! {
        <div>
            <p>"status: " {status}</p>

            <button on:click=send_message disabled=move || !connected()>"Send"</button>
            <button on:click=open_connection disabled=connected>"Open"</button>
            <button on:click=close_connection disabled=move || !connected()>"Close"</button>
            {

              move || match message.get() {
                    Some(result_info) => {
                        view! {
                            <p>"Receive message: " {format!("{:?}", result_info)}</p>
                            {result_info.result_info.trade_record.iter().enumerate().map(|(idx, _)| {
                                    view! { <li>{idx}</li> }
                                }).collect_view()}
                        }.into_any()
                    }
                    _ =>
                        // 当 message.get() 为 None 时输出空的 <p></p>
                        view! { <p></p> }.into_any()
                }
            }
        </div>
    }

**Comment by gbj** [comment]
I'm sorry, I don't understand the question.

Reading the code, though, this does not appear to be a valid Rust expression:
```rust
move || if message.get() {
                Some(result_info) => {
                    view! {
                        <p>"Receive message: " {format!("{:?}", result_info)}</p>
                        result_info.result_info.trade_record.iter().enumerate().map(|(idx, _)| {
                                view! { <li>{idx}</li> }
                            }).collect_view()
                    }.into_any()
                }
                else 
                    // 当 message.get() 为 None 时输出空的 <p></p>
                    view! { <p></p> }.into_any()
            }
        }
```
I see `if message.get() { Some(result_info) => {` which looks like a mix of an `if` expression and a match branch without a `match`

And then this is not a valid `view` macro, as Rust expressions need to be in brackets
```rust
view! {
                        <p>"Receive message: " {format!("{:?}", result_info)}</p>
                        result_info.result_info.trade_record.iter().enumerate().map(|(idx, _)| {
                                view! { <li>{idx}</li> }
                            }).collect_view()
                    }.into_any()
```
should be 
```rust
view! {
                        <p>"Receive message: " {format!("{:?}", result_info)}</p>
                        {result_info.result_info.trade_record.iter().enumerate().map(|(idx, _)| {
                                view! { <li>{idx}</li> }
                            }).collect_view()}
                    }.into_any()
```

**Reply by prehonor** [reply]
> I'm sorry, I don't understand the question.
> 
> Reading the code, though, this does not appear to be a valid Rust expression:
> 
> ```rust
> move || if message.get() {
>                 Some(result_info) => {
>                     view! {
>                         <p>"Receive message: " {format!("{:?}", result_info)}</p>
>                         result_info.result_info.trade_record.iter().enumerate().map(|(idx, _)| {
>                                 view! { <li>{idx}</li> }
>                             }).collect_view()
>                     }.into_any()
>                 }
>                 else 
>                     // 当 message.get() 为 None 时输出空的 <p></p>
>                     view! { <p></p> }.into_any()
>             }
>         }
> ```
> 
> I see `if message.get() { Some(result_info) => {` which looks like a mix of an `if` expression and a match branch without a `match`
> 
> And then this is not a valid `view` macro, as Rust expressions need to be in brackets
> 
> ```rust
> view! {
>                         <p>"Receive message: " {format!("{:?}", result_info)}</p>
>                         result_info.result_info.trade_record.iter().enumerate().map(|(idx, _)| {
>                                 view! { <li>{idx}</li> }
>                             }).collect_view()
>                     }.into_any()
> ```
> 
> should be
> 
> ```rust
> view! {
>                         <p>"Receive message: " {format!("{:?}", result_info)}</p>
>                         {result_info.result_info.trade_record.iter().enumerate().map(|(idx, _)| {
>                                 view! { <li>{idx}</li> }
>                             }).collect_view()}
>                     }.into_any()
> ```

really sorry,   I correct the question

**Reply by prehonor** [reply]
if I removed {result_info.result_info.trade_record.iter().enumerate().map(|(idx, _)| {
                                view! { <li>{idx}</li> }
                            }).collect_view()},   the code is passed

**Comment by prehonor** [comment]
view! {
        <div>
            <p>"status: " {status}</p>

            <button on:click=send_message disabled=move || !connected()>"Send"</button>
            <button on:click=open_connection disabled=connected>"Open"</button>
            <button on:click=close_connection disabled=move || !connected()>"Close"</button>

            {
              move || match message.get() {
                    Some(result_info) =>
                        {view! {
                            // <p>"Receive message: " {format!("{:?}", result_info)}</p>
                            {
                                result_info.result_info.trade_record.iter().enumerate().map(|(idx, item)| {
                                    item.iter().map(|item| {
                                        view! {
                                            <li>
                                                {item.trade_moment_info.status}
                                                {item.trade_moment_info.avg_price}
                                                {item.trade_moment_info.left_vol}
                                                {item.trade_moment_info.data_index}
                                                {item.trade_moment_info.bs_price}
                                                {item.trade_moment_info.bs_vol}
                                                {item.trade_moment_info.cur_money}
                                            </li>
                                        }
                                    }).collect_view()

                                }).collect_view()
                            }

                        }.into_view()}

                    _ =>
                        // 当 message.get() 为 None 时输出空的 <p></p>
                        {view! { <p></p> }.into_view()}
              }
            }
        </div>
    }   the code  worked
